quickjs.c

```diff
@@ -51816,6 +51816,35 @@ void JS_AddPerformance(JSContext *ctx)
     JS_FreeValue(ctx, performance);
 }
 
+/* Equality comparisons and sameness */
+JS_BOOL JS_IsLooselyEqual(JSContext *ctx, JSValue op1, JSValue op2)
+{
+    JSValue* sp = js_malloc(ctx, sizeof(JSValue) * 3);
+    sp[0] = js_dup(op1);
+    sp[1] = js_dup(op2);
+    int res = js_eq_slow(ctx, &(sp[2]), 0);
```

Comment:
```suggestion
    int res = js_eq_slow(ctx, endof(sp), 0);
```
And change sp to `JSValue sp[2];` - right now, it's heap-allocated and allocates one element too many.

Reply:
The internal API `js_eq_slow` is really not meant to be called outside the interpreter loop. We should probably add
```
static int js_eq(JSContext *ctx, JSValue op1, JSValue op2);
```
that consumes its operands and returns the boolean value and -1 for exceptions. We would use it from `js_eq_slow` just like the handlers for other relational operators.